packages/vite/src/node/config.ts

```diff
@@ -354,6 +356,15 @@ export async function resolveConfig(
     }
   }
 
+  const { publicDir } = config
+  const resolvedPublicDir =
+    publicDir === false || publicDir === ''
+      ? ''
+      : path.resolve(
+          resolvedRoot,
+          typeof publicDir === 'string' ? publicDir : 'public'
+        )
```

Comment:
**suggestion (dx):** Maybe it's easier to read the other way around

```ts
publicDir // Maybe !!publicDir is needed here to convert it to boolean
  ? path.resolve(
      resolvedRoot,
      typeof publicDir === 'string' ? publicDir : 'public'
    )
  : ''
```

Reply:
We can not do it like that because `publicDir` can be undefined, and in that case, it should resolve to 'public'

Reply:
Ah `undefined` is the case... Okay
But the with 

```ts
publicDir === false || publicDir === ''
  ? path.resolve(
      resolvedRoot,
      typeof publicDir === 'string' ? publicDir : 'public'
    )
  : ''
```

Reply:
I'm ok with both versions. Yours reads more like having `''` as the fallback value, that is not bad.